.github/workflows/ci.yml

```diff
@@ -35,10 +35,20 @@ jobs:
           toolchain: stable
           components: clippy
       - uses: actions/checkout@v1
+
+      - uses: actions/cache@v2
```

Comment:
Why though? Do you really need faster builds? This carries its own set of issues. For example, clippy does not work correctly on cached builds.

Reply:
you always want faster build when the build time is around 1 min or more, which is the case for rust projects
I wasn't aware that clippy would break with caching.

the idea was to speedup things like debian packaging (2 min) and release (4min).
the release job doesn't have a cache yet.

Reply:
https://github.com/rust-lang/rust-clippy/issues/1495

TLDR: Clippy can only analyze code that is compiled during execution of `cargo clippy`. Since files that are up to date won't be rebuilt clippy will not report errors for those.